### PR TITLE
std.os: throw compile error for `argv` on Windows

### PR DESCRIPTION
**Pull request by ominitay**
On Windows, `argv` is not populated by start code, and instead left as undefined. This is problematic, and can lead to incorrect programs compiling, but panicking when trying to access `argv`. This change causes these programs to produce a compile error on Windows instead, which is far preferable to a runtime panic.

related: #6600
cc: @enkore

### CODE REVIEW

**Comment by ominitay** [comment]
Hm, that compile error isn't very helpful in finding *where* it was referenced.

**Comment by Vexu** [comment]
Try reverting #10648

**Comment by ominitay** [comment]
> Try reverting #10648

Wow, why was that merged? That makes debugging compile errors so much harder.

**Comment by ominitay** [comment]
CI fail was the usual timeout on Windows. Will rebase and squash -- hopefully the CI decides to function correctly this time.

**Comment by ominitay** [comment]
Windows CI timed out again, another rebase until this starts working :^)

**Comment by ominitay** [comment]
Windows tests now passing :)
Ready to merge if maintainers approve

File: lib/std/os.zig
Comment (ifreund) [inline lib/std/os.zig:222]: It seems to me that this should be a switch, zig can target other platforms besides windows that don't have argv/aren't posixy. For exampl, WASI.

Comment (ominitay) [reply lib/std/os.zig:222]: Yep, it should be. I'll get that changed in a moment.

**Comment by ominitay** [comment]
Just spotted a couple of changes I'd like to make to the startup code, will get that done in a moment.

**Comment by ominitay** [comment]
Actually, it all seems fine to me. Ready to merge :)

File: lib/std/os.zig
Comment (Vexu) [inline lib/std/os.zig:223]: Looking at this again, Windows should also have `argv` if linking libc.

**Review by Vexu — CHANGES_REQUESTED** [review]

File: lib/std/os.zig
Comment (ominitay) [reply lib/std/os.zig:223]: Thanks for pointing this out :)

File: lib/std/start.zig
Comment (Vexu) [inline lib/std/start.zig:440]: This will never be called when `builtin.os.tag == .windows and !builtin.link_libc`

File: src/stage1.zig
Comment (Vexu) [inline src/stage1.zig:32]: Stage1 is always built with libc as LLVM depends on it.

**Pull request merged**
